symbolic-base/src/ZkFold/Base/Data/Vector.hs

```diff
@@ -172,3 +181,19 @@ instance (Random a, KnownNat size) => Random (Vector size a) where
 
 instance ToJSON a => ToJSON (Vector n a) where
     toJSON (Vector xs) = toJSON xs
+
+-------------------------------------------------- Algebraic instances --------------------------------------------------
+
+instance (AdditiveSemigroup a) => AdditiveSemigroup (Vector n a) where
+    (+) = zipWith (+)
+
+instance (Scale b a) => Scale b (Vector n a) where
```

Comment:
Didn't we have a generic instance for Functors?

Reply:
There is no such instance currently. @TurtlePU

Reply:
> Didn't we have a generic instance for Functors?

We did, but this instance caused coherence/overlapping errors when one had a generic type like `f a` which was actually not a functor so I removed it